api/schema.ts

```diff
@@ -0,0 +1,231 @@
+import _ from 'underscore';
+
+import {
+  Academic, APIResponseAcademic, APIResponseBlog, APIResponseEvent,
+  APIResponsePartner, Blog, CategoryResponse,
+  DataAttributes, Event, Member, MemberResponse, Partner
+} from './schemaTypes';
+import { CustomFetch } from './useFetch';
+
+const academics = (customFetch: CustomFetch) =>
+  ({
+    allAcademicCategoriesGet: async () => {
+      const res = await customFetch('GET', 'CMS', 'academic-categories');
+      const parsedCategories = (res.data as DataAttributes<CategoryResponse>)
+        .data.map(({ attributes }) => attributes.type);
+      return parsedCategories;
+    },
+    allAcademicsGet: async () => {
+      const res = await customFetch('GET', 'CMS', 'academics?populate=*');
+      const response: APIResponseAcademic = res.data;
+      const parsedAcademics: Record<number, Academic> = {};
+
+      response.data.forEach(
+        ({
+          id,
+          attributes: {
+            title,
+            creator,
+            cover_image,
+            content,
+            categories,
+            featured,
+            updatedAt,
+            description
+          },
+        }) => {
+          const parsedCategories = !_.isEmpty(categories.data)
+            ? categories.data.map(({ attributes }) => attributes.type)
+            : ['Other'];
+
+          parsedAcademics[id] = {
+            title,
+            creator,
+            coverImageUrl: `${process.env.NEXT_PUBLIC_API_URL}${cover_image.data.attributes.url}`,
+            content: content.replaceAll('/uploads/', `${process.env.NEXT_PUBLIC_API_URL}/uploads/`),
+            categories: parsedCategories,
+            featured,
+            updatedDatetime: updatedAt,
+            description,
+          };
+        }
+      );
+      return parsedAcademics;
+    },
```

Comment:
we dont want to mutate the response data in the schema file, its just to define the endpoints and type contracts for the request and responses. we do this in the hooks themselves, heres an example hook that im currently adding to the deerhacks app

![image](https://github.com/utmmcss/mcss-website-frontend/assets/70813519/be18507d-5b8b-4d3c-afe1-d2d572cedd19)

heres what the dh schema looks like for reference as of now

![image](https://github.com/utmmcss/mcss-website-frontend/assets/70813519/95b86865-65cb-4ce2-8002-dd471bb768be)